src/audio/mfcc/mfcc_setup.c

```diff
@@ -297,33 +292,17 @@ int mfcc_setup(struct processing_module *mod, int max_frames, int sample_rate, i
 
 	comp_dbg(dev, "mfcc_setup(), done");
 	return 0;
-
-free_dct_matrix:
-	rfree(state->dct.matrix);
-
-free_melfb_data:
-	rfree(fb->data);
-
-free_fft_out:
-	rfree(fft->fft_out);
-
-free_fft_buf:
-	rfree(fft->fft_buf);
-
-free_buffers:
-	rfree(state->buffers);
-
-exit:
-	return ret;
 }
 
-void mfcc_free_buffers(struct mfcc_comp_data *cd)
+void mfcc_free_buffers(struct processing_module *mod)
 {
-	fft_plan_free(cd->state.fft.fft_plan);
-	rfree(cd->state.fft.fft_buf);
-	rfree(cd->state.fft.fft_out);
-	rfree(cd->state.buffers);
-	rfree(cd->state.melfb.data);
-	rfree(cd->state.dct.matrix);
-	rfree(cd->state.lifter.matrix);
+	struct mfcc_comp_data *cd = module_get_private_data(mod);
+
+	mod_fft_plan_free(mod, cd->state.fft.fft_plan);
+	mod_free(mod, cd->state.fft.fft_buf);
+	mod_free(mod, cd->state.fft.fft_out);
+	mod_free(mod, cd->state.buffers);
+	mod_free(mod, cd->state.melfb.data);
+	mod_free(mod, cd->state.dct.matrix);
+	mod_free(mod, cd->state.lifter.matrix);
```

Comment:
`mfcc_free_buffers()` is only called from `mfcc_free()`. So we don't need explicit `mod_free()` calls? Also `mod_fft_plan_free()` only calls `mod_free()` internally